### PR TITLE
Updated react/event-loop to stable version

### LINKED ISSUE #130: Installation React Loop package version
Getting the following error when installing geo tools:

` Problem 1
    - Can only install one of: react/event-loop[v1.1.0, 0.4.x-dev].
    - Can only install one of: react/event-loop[0.4.x-dev, v1.1.0].
    - Can only install one of: react/event-loop[0.4.x-dev, v1.1.0].
    - league/geotools 0.8.0 requires react/event-loop 0.4.* -> satisfiable by react/event-loop[0.4.x-dev].
    - Installation request for league/geotools ^0.8 -> satisfiable by league/geotools[0.8.0].
    - Installation request for react/event-loop 1.1.0 -> satisfiable by react/event-loop[v1.1.0].
`

Seems geotools is using a very old version of the react-loop which makes it impossible to be used by latest API Platform.

### PR DESCRIPTION
**Pull request by stephanvierkant**
Fixes #130.

* I've updated react/event-loop to stable version.
* Dropped support for unmaintainted PHP versions, updated travis config accordingly

And some other small improvements:
- Removed unused imports
- Fixed typos in documentation
- FQCN in unit tests
- Added ext-json as a requirement

### CODE REVIEW

**Comment by stephanvierkant** [comment]
I don't know how to fix "Function ReflectionType::__toString() is deprecated". Anyone ideas?

File: composer.json [hunk 1]
```diff
@@ -22,17 +22,18 @@
     },
 
     "require" : {
-        "php"                     : "^7.0",
+        "php"                     : "^7.2",
```
Comment (oleg-andreyev) [inline composer.json:25]: I would not bump PHP version in the same PR.

**Comment by oleg-andreyev** [comment]
> I don't know how to fix "Function ReflectionType::__toString() is deprecated". Anyone ideas?

Use `\ReflectionType::getName()`

**Comment by stephanvierkant** [comment]
> > I don't know how to fix "Function ReflectionType::__toString() is deprecated". Anyone ideas?
> 
> Use `\ReflectionType::getName()`

Let me rephrase my question: what should be changed in this PR to prevent the checks from failing?

**Comment by oleg-andreyev** [comment]
@stephanvierkant so it wasn't that easy to reproduce...
This expectation is thrown by `symfony/phpunit-bridge:3.x` which is not an obvious dependency in composer.json

Ref.: https://symfony.com/doc/4.4/bundles/best_practices.html#continuous-integration

This issue is caused by \PHPUnit\Framework\MockObject\Generator::getMethodParameters in PhpUnit 6.5.5

Also note that `weak_vendors` for `SYMFONY_DEPRECATIONS_HELPER` is deprecated for `phpunit-bridge:4.x`

If you'll install `phpunit-bridge:5.x` this exception is muted.
https://github.com/symfony/phpunit-bridge/blob/master/DeprecationErrorHandler/Deprecation.php#L158

Another thing
image

**Comment by stephanvierkant** [comment]
I still don't understand what your suggestion is how to fix the test.
